### PR TITLE
Use requests rather than urllib3

### PR DESCRIPTION
**Pull request by seizethedave**
I have been doing some research around the cert verification topic, and I came to the following points:
1. mixpanel-python should strive to work without certificate hassles as much as possible. It is used in a wide variety of environments/servers/personal computers/diff't Python versions/etc.
2. I asked in the urllib3 discord channel if:
   1. if I should begin to depend on `urllib3[secure]` rather than plain old `urllib3`. The answer was "no," as that extra option is being deprecated, and will do kinda "the wrong thing" w/r/t pyOpenSSL.
   2. if I should apply this fix from `requests`  to my project: https://github.com/psf/requests/pull/5443. The answer was "yes," that it would probably help.
   3. if I should explicitly depend on `certifi`. (Included in `urllib3[secure]`, but we aren't using that.) The answer was "yes." It will remove a lot of variability in root certs in the wild. (And indeed make having a local cert store optional.)

> The current recommendation is that on modern Python versions (2.7.9+/3.4+) you're better off using the builtin Python ssl module because there's native SNI support. So that means not installing the [secure] extra and not injecting pyOpenSSL on those versions.
> It's unfortunate that the old way of doing things was the default for a lot longer than it had to be and that the extra is named "secure", we're paying the distributed tech debt price by removing it as the default and trying to document this better.
> Let me know if you have more questions :) I think the approach in the PR you mentioned is a good one but might be even better to force users to at least install Python 2.7.9+. Versions before that are really really old and insecure.

So I began to add items ii and iii to mixpanel-python. But then realized `requests` does both of those OOTB, and does a bunch of other sniffing around certifi/default certs/unzipping cert bundles/overriding certifi/etc. So to support point 1, I am just going to use `requests`.

### CODE REVIEW

File: mixpanel/__init__.py [hunk 1]
```diff
@@ -594,38 +596,36 @@ def send(self, endpoint, json_message, api_key=None, api_secret=None):
...
+                auth=basic_auth,
+                timeout=self._request_timeout,
+                verify=self._verify_cert,
             )
         except Exception as e:
             six.raise_from(MixpanelException(e), e)
 
         try:
-            response_dict = json.loads(response.data.decode('utf-8'))
+            response_dict = response.json()
```
Comment (tdumitrescu) [inline mixpanel/__init__.py:628]: requests takes care of encoding issues?

File: test_mixpanel.py [hunk 2]
```diff
@@ -449,37 +450,49 @@ class TestConsumer:
...
-            self.consumer.send('events', '{"foo":"bar"}')
-
+        responses.add(
+            responses.POST,
+            'https://api.mixpanel.com/track',
+            json={"status": 1, "error": None},
+            status=200,
+            match=[responses.json_params_matcher({"ip": 0, "verbose": 1, "data": '{"foo":"bar"}'})],
+        )
+        self.consumer.send('events', '{"foo":"bar"}')
```
Comment (tdumitrescu) [inline test_mixpanel.py:462]: I haven't used `responses` before and couldn't find it in the docs - looking at this test case, there isn't any obvious assertion that the registered response was actually called. If `self.consumer.send` doesn't make any network requests at all, will this test fail? Is that what the `activate` decorator is supposed to do?

(I do see at https://github.com/getsentry/responses#assertions-on-declared-responses "When used as a context manager, Responses will, by default, raise an assertion error if a url was registered but not accessed" but don't see anything in the readme about the decorator behavior)

Comment (seizethedave) [reply test_mixpanel.py:462]: Oh yeah, you are correct. Thx.

File: mixpanel/__init__.py (hunk 1 above)
Comment (seizethedave) [reply mixpanel/__init__.py:628]: Yep, auto-sleuthed from the response headers. https://github.com/psf/requests/blob/1466ad713cf84738cd28f1224a7ab4a19e50e361/requests/models.py#L893

Comment (seizethedave) [reply mixpanel/__init__.py:628]: wrong link: https://github.com/psf/requests/blob/1466ad713cf84738cd28f1224a7ab4a19e50e361/requests/models.py#L844-L854

**Pull request merged**